### DISCUSSION TITLE
Uniform markups for the memory requirements of structs declared in DataTypes.sol

### DISCUSSION THREAD

**Discussion by IaroslavMazur** [sablier-labs/evm-monorepo#552] [Ideas]
As of now, only the structs that are known to later be initialized and kept in contract `storage` have the information about how much memory (slots-wise) they occupy. I suggest using the term "word"/"EVM word" (instead of "slot") and having this information for all of the structs in `DataTypes.sol`.

The advantages of this approach are:
1. the increased code readability and
2. the removed burden of having to have the knowledge of the specific uses of the structs declared in `DataTypes.sol`, which increases the coupling between the declarations and the definitions of the structs (i.e. the exact opposite of the motivation for the existence of `DataTypes.sol`).

**Comment by PaulRBerg** [comment]
Good idea, thanks for taking the time to write the suggestion.

**Comment by PaulRBerg** [comment]
It turns out that we cannot use the term "word" to refer to storage slots, because "word" refers only to memory and stack variables. See my conversation with ChatGPT4 for more details:

https://chat.openai.com/share/89718543-c077-40f0-9031-15b4b78a451c

What we could do is add the term "word" in all structs that are used only in memory, but this may also lead to a confusing state of affairs.

**Reply by IaroslavMazur** [reply]
Having the memory layout of the structs described in terms of both "slots" and "words" would, indeed, be confusing.

However, when just the "slots" part is described, it's, arguably, even more confusing.

**Reply by PaulRBerg** [reply]
There's no perfect solution.

We might just want to add a caveat comment at the top of the file to explain why some structs have the "slot N" comments, and others do not.

**Reply by IaroslavMazur** [reply]
Yes, none of these solutions is too good.

Feel free to do as you please 🤗

**Reply by PaulRBerg** [reply]
Done via #581.
